### DISCUSSION TITLE
Option to require a motion distance threshold?

### DISCUSSION THREAD

**Discussion by RexWheeler** [blakeblackshear/frigate#8674] [Ideas]
I am on 0.13 beta 5, but I think this question is more general. My question is similar to #8669 

I have issues with static things in my images (rocks, plants, crap in my garage, etc.) being recognized as objects (people, dogs, etc.). Usually it’s light dependent – under certain lighting conditions, a rock or a stack of boxes becomes a person.

I know that I can set up object masks to cover these static things, however in the case of my garage, I would have to update the masks whenever I move stuff around.
 
I know that I can set size thresholds for an object to be detected, but I have cases where the false positive is close to the camera and I would want to detect a “smaller” person in the distance.

I also know that I can set up required zones to ensure that something passes through a given zone before triggering an event. While I can split my image into two or three zones so that static objects won’t be in more than one zone, any moving object that solely stays within a zone, won’t trigger an event. Careful zone creation can cover a lot of use cases, but it’s complicated.

It appears that the stationary object tracking stuff is about optimizing the recognition load for objects that are no longer moving – it doesn’t seem to be about ensuring that a recognized object is moving.

Is there a way to trigger an event only when an object is recognized _and it moves some threshold distance_? As an example I would like to be able to say, if you detect a person and start tracking it, only raise an event (retain a recording) if that person moves 20% across the frame of the image. Since mis-recognized static things don’t move (or don’t move much as the bounding boxes bounce around), having a motion threshold could simplify the filtering of false object recognition.

**Comment by uscan27** [comment]
It would be great to be able to define min/max object areas upon zones.
